server/src/main/java/org/apache/kafka/server/config/ZkConfig.java

```diff
@@ -0,0 +1,140 @@
+    public static final String ZK_SSL_KEY_STORE_PASSWORD_DOC;
+    public static final String ZK_SSL_KEY_STORE_TYPE_DOC;
+    public static final String ZK_SSL_TRUST_STORE_LOCATION_DOC;
+    public static final String ZK_SSL_TRUST_STORE_PASSWORD_DOC;
+    public static final String ZK_SSL_TRUST_STORE_TYPE_DOC;
+    public static final String ZK_SSL_PROTOCOL_DOC;
+    public static final String ZK_SSL_ENABLED_PROTOCOLS_DOC;
+    public static final String ZK_SSL_CIPHER_SUITES_DOC;
+    public static final String ZK_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_DOC;
+    public static final String ZK_SSL_CRL_ENABLE_DOC;
+    public static final String ZK_SSL_OCSP_ENABLE_DOC;
+
+    public static final int ZK_SESSION_TIMEOUT_MS = 18000;
+    public static final boolean ZK_ENABLE_SECURE_ACLS = false;
+    public static final int ZK_MAX_IN_FLIGHT_REQUESTS = 10;
+    public static final boolean ZK_SSL_CLIENT_ENABLE = false;
+    public static final String ZK_SSL_PROTOCOL = "TLSv1.2";
+    public static final String ZK_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM = "HTTPS";
+    public static final boolean ZK_SSL_CRL_ENABLE = false;
+    public static final boolean ZK_SSL_OCSP_ENABLE = false;
+
+    static {
+        Map<String, String> zkSslConfigToSystemPropertyMap = new HashMap<>();
+
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_CLIENT_ENABLE_PROP, ZKClientConfig.SECURE_CLIENT);
+        zkSslConfigToSystemPropertyMap.put(ZK_CLIENT_CNXN_SOCKET_PROP, ZKClientConfig.ZOOKEEPER_CLIENT_CNXN_SOCKET);
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_KEY_STORE_LOCATION_PROP, "zookeeper.ssl.keyStore.location");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_KEY_STORE_PASSWORD_PROP, "zookeeper.ssl.keyStore.password");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_KEY_STORE_TYPE_PROP, "zookeeper.ssl.keyStore.type");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_TRUST_STORE_LOCATION_PROP, "zookeeper.ssl.trustStore.location");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_TRUST_STORE_PASSWORD_PROP, "zookeeper.ssl.trustStore.password");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_TRUST_STORE_TYPE_PROP, "zookeeper.ssl.trustStore.type");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_PROTOCOL_PROP, "zookeeper.ssl.protocol");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_ENABLED_PROTOCOLS_PROP, "zookeeper.ssl.enabledProtocols");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_CIPHER_SUITES_PROP, "zookeeper.ssl.ciphersuites");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_PROP, "zookeeper.ssl.hostnameVerification");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_CRL_ENABLE_PROP, "zookeeper.ssl.crl");
+        zkSslConfigToSystemPropertyMap.put(ZK_SSL_OCSP_ENABLE_PROP, "zookeeper.ssl.ocsp");
+
+        ZK_SSL_CONFIG_TO_SYSTEM_PROPERTY_MAP = Collections.unmodifiableMap(zkSslConfigToSystemPropertyMap);
+
+        ZK_SSL_CLIENT_ENABLE_DOC = "Set client to use TLS when connecting to ZooKeeper." +
```

Comment:
Done

Reply:
I reverted those changes and leave my first version.
Because, java8 fails to load class if we use map values inside String constants definition.

https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-15075/16/pipeline/9 

```
[2024-03-26T15:45:33.431Z] java.lang.StackOverflowError
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.comp.Attr.visitLambda(Attr.java:2296)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.tree.JCTree$JCLambda.accept(JCTree.java:1624)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.comp.Attr.visitLambda(Attr.java:2435)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.tree.JCTree$JCLambda.accept(JCTree.java:1624)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.comp.Attr.visitLambda(Attr.java:2435)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.tree.JCTree$JCLambda.accept(JCTree.java:1624)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.comp.Attr.visitLambda(Attr.java:2435)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.tree.JCTree$JCLambda.accept(JCTree.java:1624)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.comp.Attr.visitLambda(Attr.java:2435)
[2024-03-26T15:45:33.431Z] 	at com.sun.tools.javac.tree.JCTree$JCLambda.accept(JCTree.java:1624)
```

Reply:
oh, nice finding. please ignore my previous comment :)